### PR TITLE
Matmul Mul ops

### PR DESCRIPTION
**Pull request by justin1121**
Plus bonus c++ linting based off google style guide.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,4 +7,7 @@ test: .bazelrc
 fmt:
 	cd tf_big && find . -iname *.h -o -iname *.cc | xargs clang-format -i -style=google
 
+lint:
```
Comment (mortendahl) [inline Makefile]: 😍

File: tf_big/cc/kernels/big_tensor.h [hunk 2]
```diff
@@ -45,11 +51,10 @@ typedef Matrix<mpz_class, Dynamic, Dynamic> MatrixXm;
 namespace tf_big {
 
 struct BigTensor {
- public:
-  BigTensor(){};
+  BigTensor() {}
   BigTensor(const BigTensor& other);
-  BigTensor(mpz_class m);
-  BigTensor(const MatrixXm& mat);
+  explicit BigTensor(mpz_class m);
```
Comment (mortendahl) [inline tf_big/cc/kernels/big_tensor.h]: what does `explicit` do?

File: tf_big/cc/kernels/big_kernels.cc [hunk 3]
```diff
@@ -81,11 +82,49 @@ class BigAddOp : public OpKernel {
     Tensor* output;
     OP_REQUIRES_OK(ctx, ctx->allocate_output(0, TensorShape{}, &output));
 
-    MatrixXm res = val1->value + val2->value;
-    BigTensor big(res);
+    auto res = *val1 + *val2;
```
Comment (mortendahl) [inline tf_big/cc/kernels/big_kernels.cc]: how is data moved around here?

File: tf_big/python/ops/big_ops_test.py [hunk 4]
```diff
@@ -46,21 +48,57 @@ def test_add(self):
...
+      output = sess.run(c_str)
+
+      np.testing.assert_equal(output, expected)
+
+  def test_matmul(self):
+    with tf.Session() as sess:
+      a = np.array([[5, 5], [5, 5]]).astype(np.int32)
+      b = np.array([[6, 6], [6, 6]]).astype(np.int32)
+
+      expected = a.dot(b)
```
Comment (mortendahl) [inline tf_big/python/ops/big_ops_test.py]: `dot` is from Eigen? TF seems to exclusively use `matmul`

File: tf_big/cc/kernels/big_tensor.h (hunk 2 above)
Comment (justin1121) [reply tf_big/cc/kernels/big_tensor.h]: https://google.github.io/styleguide/cppguide.html#Implicit_Conversions, basically just makes sure the types passed into the constructor aren't implicitly converted.

File: tf_big/cc/kernels/big_kernels.cc (hunk 3 above)
Comment (justin1121) [reply tf_big/cc/kernels/big_kernels.cc]: Can you rephrase your question? Not sure what you mean.

File: tf_big/python/ops/big_ops_test.py (hunk 4 above)
Comment (justin1121) [reply tf_big/python/ops/big_ops_test.py]: this is just numpy, maybe it should say `matmul`?

Comment (mortendahl) [reply tf_big/python/ops/big_ops_test.py]: ah right, nvm then :)

**Comment by justin1121** [comment]
@mortendahl I'm going to merge this up. Feel free to follow up on any of the comments above.
